samples/helloworld/README.md

```diff
@@ -508,6 +510,18 @@ ninja
 ninja run
 ```
 
+### Mixing LVI Mitigation
+
+You can use a different LVI mitigation method to build your app than Open Enclave, however additional cmake configuration may be required. See the table below for all possible combinations.
+
+
+| OE    | client | Example cmake command                                                                                   | Additional Requirement                                                                                                   |
+|-------|-----------|---------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------|
+| GNU   | GNU     | `cmake .. -DLVI_MITIGATION=ControlFlow-GNU -DLVI_MITIGATION_BINDIR=/path/to/lvi_mitigation_bin`           |                                                                                                                          |
+| GNU   | Clang       | `cmake .. -DLVI_MITIGATION=ControlFlow-Clang -DLVI_MITIGATION_BINDIR=/path/to/lvi_mitigation_bin`         | Must pass `OE_USED_CONTROLFLOW_GNU` to `apply_lvi_mitigation`. Example: `apply_lvi_mitigation(enclave, OE_USED_CONTROLFLOW_GNU)` |
```

Comment:
Since we always use OE_USED_CONTROLFLOW_GNU, can we make it the default option. We are not releasing packages with controlflow_clang.
I would remove this table and make the user experience - OE built with controlflow_gnu, in apply_lvi_mitigation, assume that controlflow_gnu was used, unless otherwise specified.

Reply:
Yes, that sounds more user-friendly. Will do

Reply:
The commit 08f3fb51d2c444d94098223c3d1a708d61b79b8e address this comment.

A few changes:
1. now we have 2 copies of `apply_lvi_mitigation` that are largely the same, one for building Open Enclave, which doesn't need optional arg to change its behavior, the other provided specifically to clients to build their project, which accepts optional arg
2. For the `apply_lvi_mitigation` used by the client, it defaults to assume OE is built with GNU, and usage remains the same. If OE and client project are both built by clang, then need to pass in `NO_CUSTOM_TOOLCHAIN`

Reply:
Now the command example looks like this
| OE    | client | Supported | Example cmake command                                                                                   | Additional Requirement                                                                                                   |
|-------|--------|-----------|---------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------|
| GNU   | GNU    | Yes       | `cmake .. -DLVI_MITIGATION=ControlFlow-GNU -DLVI_MITIGATION_BINDIR=/path/to/lvi_mitigation_bin`           |                                                                                                                          |
| GNU   | Clang  | Yes       | `cmake .. -DLVI_MITIGATION=ControlFlow-Clang -DLVI_MITIGATION_BINDIR=/path/to/lvi_mitigation_bin`         |  |
| Clang | GNU    | Yes       | `cmake .. -DLVI_MITIGATION=ControlFlow-GNU -DLVI_MITIGATION_BINDIR=/path/to/lvi_mitigation_bin`           |                                                                                                                          |
| Clang | Clang  | Yes       | `cmake .. -DLVI_MITIGATION=ControlFlow-Clang -DCMAKE_C_COMPILER=clang-11 -DCMAKE_CXX_COMPILER=clang++-11` |      Must pass `NO_CUSTOM_TOOLCHAIN` to `apply_lvi_mitigation`. Example: `apply_lvi_mitigation(enclave, NO_CUSTOM_TOOLCHAIN)`    |